tgui/packages/tgui/interfaces/TramController.tsx

```diff
@@ -0,0 +1,279 @@
+import { useBackend, useLocalState } from '../backend';
+import { BooleanLike } from 'common/react';
+import { Section, NoticeBox, Dropdown, LabeledList, ProgressBar, Flex, Button } from '../components';
+import { toFixed } from 'common/math';
+import { Window } from '../layouts';
+
+type TransportData = {
+  transportId: string;
+  controllerActive: number;
+  controllerOperational: BooleanLike;
+  travelDirection: number;
+  destinationPlatform: string;
+  idlePlatform: string;
+  recoveryMode: BooleanLike;
+  currentSpeed: number;
+  currentLoad: number;
+  statusSF: boolean;
+  statusCE: boolean;
+  statusES: boolean;
+  statusPD: boolean;
+  statusDR: boolean;
+  statusCL: boolean;
+  statusBS: boolean;
+  destinations: TramDestination[];
+};
+
+type TramDestination = {
+  name: string;
+  dest_icons: string[];
+  id: number;
+};
+
+export const TramController = (props, context) => {
+  const { act, data } = useBackend<TransportData>(context);
+
+  const {
+    transportId,
+    controllerActive,
+    controllerOperational,
+    travelDirection,
+    destinationPlatform,
+    idlePlatform,
+    recoveryMode,
+    currentSpeed,
+    currentLoad,
+    statusSF,
+    statusCE,
+    statusES,
+    statusPD,
+    statusDR,
+    statusCL,
+    statusBS,
+    destinations = [],
+  } = data;
+
+  const [tripDestination, setTripDestination] = useLocalState(
+    context,
+    'TramDestination',
+    ''
+  );
+
+  return (
+    <Window title="Tram Controller" width={800} height={325} theme="dark">
+      <Window.Content>
+        <Flex direction="row">
```

Comment:
Medium sized ask- switch these to stack, stack item respectively. Instead of using pixel measurements for H/W dimensions, use grow property, ie,

```ts
// this would take 1/3 the space
<Stack.Item grow={1}
// this would take 2/3 the space
<Stack.Item grow={2}
```

Reply:
Sure

Reply:
Sure.